universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -120,16 +120,10 @@ private void putNull(Path path) {
    * @param value the value to place; values of type Map will create the equivalent JSON structure
    */
   private void put(Path path, Object value) {
-    path.parentPaths()
-        .forEach(
-            segmentPath -> {
-              if (!hasPath(segmentPath)) {
-                this.jsonData.put(
-                    segmentPath.parentPath().path(), segmentPath.keyName(), new HashMap<>());
-              }
-            });
-
-    this.jsonData.put(path.parentPath().path(), path.keyName(), value);
+    if (!hasPath(path.parentPath())) {
```

Comment:
Neat!

I know "applicant" is pretty much in every path, but I think we should still include a base case here. If for some reason the path doesn't start with "applicant", eventually you'll get to Path.empty(), which will throw an error like "Json cannot be null or empty"

Reply:
Ahh thanks! I think that may be why all the other tests are failing.

Reply:
It was not! :(